### PR TITLE
[TheiaCoV] Split database from Kraken2_TheiaCoV task

### LINKED ISSUE #520: [TheiaCoV] Update kraken2 viral database to fix weird RSV-B nomenclature
:bug:

### :pencil: Describe the Issue

Copied from #436 

> set 2 new default Strings for kraken_target_organism String rsv_a_kraken_target_organism = "Respiratory syncytial virus" and > String rsv_b_kraken_target_organism = "Human orthopneumovirus"
⚠️ NOTE: this was done due to the old kraken2 database used in this container us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.0.8-beta_hv that is used in the kraken2_theiacov WDL task. We can revisit updating this task in the future. It’s pretty central to a lot of workflows so it would be a big change to update the database with up-to-date NCBI taxonomy IDs embedded within the kraken2 database. NCBI taxonomy has been updated so that Human orthopneumovirus is equivalent to human respiratory syncytial virus: https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=11250&lvl=3&lin=f&keep=1&srchmode=1&unlock

### :repeat: How to Reproduce

- Was this run on the Terra platform?
  - Terra on GCP or Azure?
- Or were the workflows run locally using `miniwdl` or `cromwell`?
  - **What was the exact command was used to launch the workflow?**
- If possible, we may ask you to share your Terra workspace with us. Usually READER access is sufficient, but we may ask for WRITER access if we need to make changes to the workspace to reproduce the issue.

### LINKED ISSUE #563: [TheiaCoV] Split kraken2 viral database from kraken2_theiacov task
:cool:

### :pushpin: Explain the Request

This would allow the underlying code and the database to be updated independently, leading to better sustainability! 

https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/contamination/task_kraken2.wdl#L13

### PR DESCRIPTION
**Pull request by cimendes**
This PR closes #563, closes #520

🗑️ This dev branch should <NOT> be deleted after merging to main.

## :brain: Summary

This PR addresses several issues that came up when the RSV target organism was set differently for RSV-A and RSV-B, i.e. `rsv_a_kraken_target_organism = "Respiratory syncytial virus"` and `rsv_b_kraken_target_organism = "Human orthopneumovirus"` due to the use of an old database. As the existing docker image had an embedded database, it meant that each time the Kraken database would need to be updated, it would have to be done inside the docker container, which would not be efficient. 

This PR simply updates the existing `kraken2_theiacov` task to split the database from the container. The new database is custom built with human + refseq viral sequences as of 2024-08-28.

No inputs or outputs have been altered other than what is strictly required to make the new code changes functional.

Behaviour change:
- the percentage of sc2 is now only reported for TheiaCoV is the target organism is indeed SC2. To account for this the output type has been changed from Float to String.
- Freyja now has a kraken2_target_organism input that by default is SC2, but is user modifyable

## :zap: Impacted Workflows/Tasks

Impacted tasks
- kraken2_theiacov
- kraken2_parse_classified

Impacted workflows
- TheiaCoV
  - Illumina_PE
  - Illumina_SE
  - Clearlabs
  - ONT
- Freyja_FASTQ
- NCBI_Scrub
  - SE
  - PE 

This PR may lead to different results in pre-existing outputs: **Yes**
- Use of an updated database may reflect changes in how taxa are identified.

This PR uses an element that could cause duplicate runs to have different results: **No**

### :gear: Algorithm

As explained above, the only change in algorithm is that the percentage of sc2 is now only reported if the target organism is SC2, otherwise this column is blank. To accomplish this the data type has been changed from Float to String. Impacts column sorting as it is now alpha-numeric.

### ➡️ Inputs

freyja_fastq new input:
- kraken2_target_organism

### ⬅️ Outputs

Percent sc2 is now a string output, rather than a float output

## :test_tube: Testing

- Freyja_FASTQ (SC2 PE): https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/3c8f46c1-a0b7-44cb-b949-10f6323b9c9b ⚠️ 2 failures but unrelated, they are weird tho! 
- TheiaCoV
  - ClearLabs (SC2 only): https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/e917f233-897d-43b9-b6ff-1d6e2d594c61 ✅ 
  - SE (multiple orgs, including RSV-A and RSV-B): https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/ddc6bb13-c026-46ca-a8f6-4f09b0f97722 ⚠️ unrelated failures in 3 wnv samples, maybe worth investigating?
  - ONT (multiple orgs): https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/f30871bd-ebda-4133-8699-9b120d4023f7 ⚠️ failures in all HIV samples because HIV hates me, but unrelated to the PR
  - PE (multiple orgs, including RSV-A and RSV-B): https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/b5fae176-0f07-4f8e-9266-e66aaec7e7cd ⚠️ 1 failure in wnv unrelated to the PR, same error as observed in SE runs
- NCBI_Scrub
  - PE (SC2): https://app.terra.bio/#workspaces/theiagen-validations/ines_sandbox/job_history/c4360ec0-9779-440e-9982-ac10e2fc5361 ✅ 
  - SE (SC2): https://app.terra.bio/#workspaces/theiagen-validations/ines_sandbox/job_history/9c222e8c-4291-49c7-8453-011353e393f2 ✅ 
 
## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [x] All changed results have been confirmed
- [x] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [x] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments
- [x] The documentation has been updated

### CODE REVIEW

File: workflows/theiacov/updates/wf_ncbi_scrub_pe.wdl [hunk 1]
```diff
@@ -20,7 +20,8 @@ workflow dehost_pe {
     input:
       samplename = samplename,
       read1 = ncbi_scrub_pe.read1_dehosted,
-      read2 = ncbi_scrub_pe.read2_dehosted
+      read2 = ncbi_scrub_pe.read2_dehosted,
+      target_organism = "Severe acute respiratory syndrome coronavirus 2"
```
Comment (Michal-Babins) [inline workflows/theiacov/updates/wf_ncbi_scrub_pe.wdl:24]: For NCBI scrub workflows can we set this to be a workflow input and not hard coded on the task level?

File: workflows/theiacov/updates/wf_ncbi_scrub_se.wdl [hunk 2]
```diff
@@ -17,7 +17,8 @@ workflow dehost_se {
   call kraken.kraken2_theiacov as kraken2 {
     input:
       samplename = samplename,
-      read1 = ncbi_scrub_se.read1_dehosted
+      read1 = ncbi_scrub_se.read1_dehosted,
+      target_organism = "Severe acute respiratory syndrome coronavirus 2"
```
Comment (Michal-Babins) [inline workflows/theiacov/updates/wf_ncbi_scrub_se.wdl:21]: Same here for workflow level input

**Review by Michal-Babins — CHANGES_REQUESTED** [review]
For ncbi scrubber, please pass in workflow level variable and do not hard code on task level.

**Comment by Michal-Babins** [comment]
Testing illumina PE [here](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/58507ee8-c048-4129-b482-f01e6cceeb12) for WNV, MPox, Flu, Sars-Cov-2.

**Comment by Michal-Babins** [comment]
Everything looks good in the test and the output is properly identified by kraken from what we would expect. One of the flu samples returned nothing from kraken, but I ran it against PHBv2.2.1 and it was the same thing so something with that sample is just slightly off. 

Run with isolated sample [here](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/1fe43e1f-03a0-4780-8e56-9a77138b01a0)

**Comment by cimendes** [comment]
@Michal-Babins I've exposed the required input parameter, as requested! A little merge conflict has been solved and the CI has been updated.

**Comment by cimendes** [comment]
Ops, forgot about the docs update! Now everything is done!

**Review by Michal-Babins — APPROVED** [review]
Specified at the higher level now, looks good to me. Thank you.

**Pull request merged**